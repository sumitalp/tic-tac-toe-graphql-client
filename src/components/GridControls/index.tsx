import * as React from "react";
import classNames from "classnames";

import style from "./GridControls.css";

const GridControls = (props: {
    className?: string;
    isMultiplayer: boolean;
    gameUrl: string;
}) => (
    <div className={classNames(style.root, props.className)}>
        {null}
    </div>
);

export default GridControls;
