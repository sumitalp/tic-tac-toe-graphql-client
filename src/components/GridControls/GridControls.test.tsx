import * as React from "react";
import { shallow } from "enzyme";
import GridControls from "./index";

describe("GridControls", () => {
    const defaultProps = {
        isMultiplayer: false,
        gameUrl: "http://example.com/game/123"
    };

    it("should render for multiplayer", () => {
        const component = shallow(<GridControls {...defaultProps} />);

        expect(component).toMatchSnapshot();
    });

    it("should render for single player", () => {
        const component = shallow(
            <GridControls {...defaultProps} isMultiplayer={false} />
        );

        expect(component).toMatchSnapshot();
    });
});
