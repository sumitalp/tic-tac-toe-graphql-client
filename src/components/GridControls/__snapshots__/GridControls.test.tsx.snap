// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridControls should render for multiplayer 1`] = `
<div
  className="root"
/>
`;

exports[`GridControls should render for single player 1`] = `
<div
  className="root"
/>
`;
